src/libvirtApi/domain.js

```diff
@@ -137,10 +135,22 @@ export function domainAttachDisk({
     return domainAttachDevice({ connectionName, vmId, permanent, hotplug, xmlDesc });
 }
 
-export function domainAttachIface({ connectionName, vmId, mac, permanent, hotplug, sourceType, source, model }) {
-    const xmlDesc = getIfaceXML(sourceType, source, model, mac);
+export function domainAttachIface({ connectionName, vmName, mac, permanent, hotplug, sourceType, source, model }) {
+    const options = { err: "message" };
+    if (connectionName === "system")
+        options.superuser = "try";
 
-    return domainAttachDevice({ connectionName, vmId, permanent, hotplug, xmlDesc });
+    let update = "";
+    if (hotplug)
+        update = "--update";
+    let define = "--define";
+    if (hotplug && !permanent)
+        define = "--no-define";
+
+    return cockpit.script(
+        `virt-xml -c qemu:///${connectionName} ${vmName} --add-device --network mac=${mac},type=${sourceType},source=${source},model=${model} ${define} ${update}`,
```

Comment:
When 'generate automatically' is chosen for the MAC the mac variable is undefined and clicking add would result in an error:
```  ERROR XML error: unable to parse mac address 'undefined'  ```